sdk/core/azure-core-management/src/main/java/com/azure/core/management/implementation/polling/PollingState.java

```diff
@@ -170,7 +170,7 @@ public static PollingState from(SerializerAdapter serializerAdapter, String valu
     /**
      * @return the current status of the long-running-operation.
      */
-    LongRunningOperationStatus getOperationStatus() {
+    public LongRunningOperationStatus getOperationStatus() {
```

Comment:
@yaohaizh - consumers are suppose to call getStatus() on AsyncPollRrsponse,PollRrsponse which returns LongRunningStatus.  Are we making LongRunningOperation::getOperationStatus() public for any internal (within mgmt sdk) use?

Reply:
There is the first factory method in this file that PollState is not set internally in the PollerFactory file, thus, our MGMT code is such as https://github.com/Azure/azure-libraries-for-java/commit/09e3cfad5fa3800ab7b746ffed1e5513df7d25e9#diff-2cfa33e57f52c488063ba39769edcac8R131

Reply:
I try to put above code into PollerFactory to totally handle PollState, but it seems not easy given the current method signature.

Reply:
Ok got it, so to enable synchronous completion scenario: along with the [feature](https://github.com/Azure/azure-sdk-for-java/pull/7599) you requested, we also need `PollingState:: getOperationStatus ()` to be accessible. This should be fine - given `PollingState` is in impl package and customers are not suppose to use it directly. 

After couple of lroInitOperation use cases for various services, if we find we can generalize init op then we could have a version of factory with default initOp baked-in.

Reply:
Agree, current the design is not mature since it manipulates the PollState in different places. Ideally, the acess level change should be avoided. 

The https://github.com/Azure/azure-libraries-for-java/blob/fix_tests/azure-mgmt-resources/src/main/java/com/azure/management/AzureServiceClient.java `getLroResultAsync` & `activationOperation` has been tested aganist ARM/storage services and works well. 

Please help generalizes these methods into the azure-core-management if you think it is more reasonable. Thanks!

Reply:
thanks for the example, how about [this](https://github.com/Azure/azure-sdk-for-java/pull/6303/commits/07f909cb446aac55c5ce4f59ec36635a2436d859)